chasm/lib/activity/handler.go

```diff
@@ -82,7 +86,22 @@ func (h *handler) StartActivityExecution(ctx context.Context, req *activitypb.St
 		chasm.WithRequestID(req.GetFrontendRequest().GetRequestId()),
 		chasm.WithBusinessIDPolicy(reusePolicy, conflictPolicy),
 	)
+
 	if err != nil {
+		var alreadyStartedErr *chasm.ExecutionAlreadyStartedError
+		if errors.As(err, &alreadyStartedErr) {
+			details := &errordetailspb.ActivityExecutionAlreadyStartedFailure{
+				StartRequestId: alreadyStartedErr.CurrentRequestID,
+				RunId:          alreadyStartedErr.CurrentRunID,
+			}
+
+			errStatus, errDetail := status.New(codes.AlreadyExists, "activity execution already started").WithDetails(details)
+			if errDetail != nil {
+				return nil, errDetail
```

Comment:
Should we actually be returning `err` here as opposed to `errDetail`?

My worry is that `errDetail` is going to contain an obscure message about something that went wrong while we were trying to construct details, whereas it is more helpful and appropriate to give the user `chasm.ExecutionAlreadyStartedError`.